deployer/helm_upgrade_decision.py

```diff
@@ -0,0 +1,358 @@
+"""
+Functions related to deciding which clusters and/or hubs need their *hub helm chart or
+support helm chart upgrading depending on an input list of filenames that have been
+added or modified in a GitHub Pull Request.
+"""
+import fnmatch
+import os
+from pathlib import Path
+from ruamel.yaml import YAML
+from rich.console import Console
+from rich.table import Table
+
+yaml = YAML(typ="safe", pure=True)
+
+# Determine if we are running a test or not. We set this env var to true in the
+# pytest.ini file so it is set when the package is tested.
+test_env = os.getenv("RUN_ENV", False)
+
+
+def discover_modified_common_files(modified_paths: list):
```

Comment:
I'm not sure what this function should be named, but I'm uneasy that it returns a critical decision if to upgrade or not and it's not something conveyed by the name.

![image](https://user-images.githubusercontent.com/3837114/158227429-ccbee4e3-d541-44e0-9b38-db94c46d7b5a.png)